core/src/main/scala/org/apache/spark/scheduler/ExecutorLossReason.scala

```diff
@@ -54,10 +54,11 @@ private [spark] object LossReasonPending extends ExecutorLossReason("Pending los
 /**
  * @param _message human readable loss reason
  * @param workerLost whether the worker is confirmed lost too (i.e. including shuffle service)
+ *                   Used to know whether the shuffle data is lost too.
  */
 private[spark]
-case class SlaveLost(_message: String = "Slave lost", workerLost: Boolean = false)
-  extends ExecutorLossReason(_message)
+case class SlaveLost(_message: String = "Slave lost", workerLost: Boolean = false,
+                     expected: Boolean = false) extends ExecutorLossReason(_message)
```

Comment:
please add comment to the new param, also is it good to rename to `causedByApp` ?